### DISCUSSION TITLE
Postprocessing Kernel Variables

### DISCUSSION THREAD

**Discussion by prajwalka** [idaholab/moose#17681] [Q&A General] [answered]
Hi,

I'm fairly new to Moose and had this question.

Is there an equivalent of "Material Property" for kernels? I wanted to use "declareProperty" for a kernel variable and postprocess it using something like MaterialRankTwoAux kernel. I also wanted the value of this variable at the previous time step.

For example, in enhanced strain methods, the value of stress and strain obtained from the linear elastic material need to be modified at the kernel level using tensors depending on the quadrature point. Is there any way of outputting these modified stress and strain values which are stored in my kernel? Or is it preferred that this modification is done using a User Object or any other method?

Thanks a lot!

**Comment by hugary1995** [comment, marked as answer]
I would recommend you to write a material property that calculates the modified stress/strain.

**Reply by aeslaughter** [reply]
You can also create a custom AuxKernel to perform a calculation that uses coupled in the solution variable. We have an example of this in our workshop presentation: 

- The slides are here https://mooseframework.inl.gov/workshop/#/28
- Our discussion about it is here: https://www.youtube.com/watch?v=2tJwBsYaLaI
